app/javascript/packs/outdoor_map_builder.js

```diff
@@ -207,12 +222,51 @@ window.routingControl = L.Routing.control({
 })
 .on('waypointschanged', (e)=>{
   console.log("waypointschanged");
+  // we only highlight the destination of the current navigation route
+  highlightDestinationDifferently(routingControl.getWaypoints()[1].latLng);
+  
   // this handler is called whenever the waypoints are changed in any way (search bar or clicking in the map)
   routingControl.show()
   // always calculate the route to show the 'A' marker if only one waypoint is set
   routingControl.route()
 });
 
+let highlightedBuilding = null
+
+function highlightDestinationDifferently(position) {
```

Comment:
der Methodenname ist eetwas suboptimal zum Verständnis, besonders in Verbindung mit der anderen Methode `setStyleForHighlightedBuilding` wird die Aufgabenteilung nicht sonderlich klar. Die Methode hier findet hauptsächtlich heraus, welches Gebäude in der gegebenen Position liegt, setzt dann die highlightedBuilding Variable auf diese und setzt dann den Style (was ein logischer Schritt ist, nachdem man das geändert hat, aber es ist halt nicht was die Methode hauptsächlich tut). Ich fänd z.B. `changeHighlightedBuilding` besser.